packages/repository/src/relations/relation.helpers.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as _ from 'lodash';
+import {Entity, EntityCrudRepository, Filter, Options, Where} from '..';
+
+/**
+ * Finds model instances that contain any of the provided foreign key values.
+ *
+ * @param targetRepository - The target repository where the model instances are found
+ * @param fkName - Name of the foreign key
+ * @param fkValues - Array of the values of the foreign keys to be included
+ * @param scope - Additional scope constraints (not currently supported)
+ * @param options - Options for the operations
+ */
+export async function findByForeignKeys<
+  Target extends Entity,
+  TargetID,
+  TargetRelations extends object,
+  ForeignKey
+>(
+  targetRepository: EntityCrudRepository<Target, TargetID, TargetRelations>,
+  fkName: StringKeyOf<Target>,
+  fkValues: ForeignKey[],
```

Comment:
Can we accept a single value too? For example:

```ts
fkValues: ForeignKey[] | ForeignKey,
```
```

Reply:
Is it possible to have a type param such as `FK extends StringKeyOf<Target>` and use it to constrain fk values, such as `Target[FK]`?

Reply:
Let me reword it to make sure that I understand the type constrains in `tsc` correctly:

if we do `foreignKey extends StringKeyOf<Target>`, it would constrain the passed in `foreignKey` to be a string-like `<Target>` instance?

Reply:
I'm proposing the following:

```ts
export async function findByForeignKeys<
  Target extends Entity,
  FK extends StringKeyOf<Target>,
  TargetRelations extends object,
>(
  targetRepository: EntityCrudRepository<Target, unknown, TargetRelations>,
  fkName: FK,
  fkValues: Target[FK][],
  scope?: Filter<Target>,
  options?: Options,
): Promise<(Target & TargetRelations)[]> {
}
```